bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v3/product/ProductImpl.java

```diff
@@ -189,14 +190,14 @@ private ArrayNode fetchVariantsAsJsonArray() throws JsonProcessingException {
 
             for (Asset asset : variant.getAssets()) {
                 ObjectNode jsonAsset = mapper.createObjectNode();
-                jsonAsset.put("path", asset.getPath());
+                jsonAsset.put("path", StringEscapeUtils.escapeHtml4(asset.getPath()));
```

Comment:
We have a preferred API for this: org.apache.sling.xss.XSSAPI . Please check the CIF add-on bundle sources for examples and see the javadoc at: https://sling.apache.org/apidocs/sling9/org/apache/sling/xss/XSSAPI.html
It highly recommended by security to use this library in AEM projects.